azure/services/dns/client.go

```diff
@@ -29,23 +28,24 @@ type azureClient struct {
 
 var _ client = (*azureClient)(nil)
 
-func newAzureClient(azureCluster *v1beta1.AzureCluster) (*azureClient, error) {
-	cred, err := azidentity.NewDefaultAzureCredential(nil)
+func newAzureClient(scope scope.DNSScope) (*azureClient, error) {
+	clientID := azidentity.ClientID(scope.ClientID())
```

Comment:
where do we define this `ClientID`  , because is important that this is the Identity of the AzureClusterIdentity used to create the cluster or it might fail for WC

I think we need
* Fetch the AzureClusterIdentity for each cluster
* check if the identity is a Managed Identity or a Static Identity ( customers could still do that if they want )
* if Managed Identity then create with NewManagedIdentityCredential using the clientId specified in the AzureclusterIdentity
* if static then create a Static credential using the secret

We have no other way to ensure that the credentials used will be able to create a DNS zone in the resource group of the Workload Cluster